inventory_requests/views/ModifyRequestCart.py

```diff
@@ -0,0 +1,131 @@
+
+    def patch(self, request, pk, format=None):
+        request_cart_to_cancel = get_or_not_found(RequestCart, pk=pk)
+        if modify_request_cart_logic.can_approve_deny_cancel_disburse_request_cart(request_cart_to_cancel, "cancelled"):
+            request_cart_to_cancel.status = "cancelled"
+            # reason is guaranteed to not be null since it is required in a request
+            if request.data.get('comment') is not None:
+                request_cart_to_cancel.reason = "{old_reason} cancellation reason is : {new_reason}"\
+                    .format(old_reason=request_cart_to_cancel.reason, new_reason=request.data.get('comment'))
+            request_cart_to_cancel.save()
+            LoggerUtility.log(initiating_user=request.user, nature_enum=ActionEnum.REQUEST_CANCELLED,
+                              affected_user=request.user, carts_affected=[request_cart_to_cancel])
+            updated_cart = RequestCart.objects.get(pk=request_cart_to_cancel.id)
+            serializer_cart = RequestCartSerializer(updated_cart)
+            return Response(data=serializer_cart.data, status=status.HTTP_200_OK)
+        else:
+            raise MethodNotAllowed(self.patch, detail="Cannot cancel request")
+
+
+class DenyRequestCart(APIView):
+    permission_classes = [IsStaffUser]
+
+    def patch(self, request, pk, format=None):
+        return approve_deny_request_cart(self, request, pk, "denied")
+
+
+class FulfilRequestCart(APIView):
+    permission_classes = [IsStaffUser]
+
+    def patch(self, request, pk):
+        request_cart = get_or_not_found(RequestCart, pk=pk)
+        if request_cart.status == 'approved':
+            request_cart.status = 'fulfilled'
+            request_cart.save()
+            LoggerUtility.log(initiating_user=request_cart.staff, nature_enum=ActionEnum.REQUEST_FULFILLED,
+                              affected_user=request_cart.owner, carts_affected=[request_cart])
+            return Response(data=RequestCartSerializer(request_cart).data, status=status.HTTP_200_OK)
+        raise MethodNotAllowed(method=self.patch, detail="Request must be approved but is currently {current_status}"
+                               .format(current_status=request_cart.status))
+
+
+class DispenseRequestCart(generics.UpdateAPIView):
+    permission_classes = [IsStaffUser]
+    serializer_class = StaffDisburseSerializer
+    queryset = RequestCart.objects.all()
+
+    def patch(self, request, *args, **kwargs):
+        request.data['status'] = 'fulfilled'
+        self.partial_update(request, *args, **kwargs)
+        updated_request = self.get_object()
+        LoggerUtility.log(initiating_user=updated_request.staff, nature_enum=ActionEnum.ITEMS_DISBURSED,
+                          affected_user=updated_request.owner, carts_affected=[updated_request])
+        return Response(data=RequestCartSerializer(updated_request).data, status=status.HTTP_200_OK)
+
+    def put(self, request, *args, **kwargs):
```

Comment:
hm i see. but in what situation would they do a put request? like if we give them the api and we make the calls ourselves for the ui, would someone ever make a put request?